lib/response_bank.rb

```diff
@@ -64,7 +64,7 @@ def cache_key_for(data)
         key = %{#{key}:#{hash_value_str(data[:version])}} if data[:version]
 
         # add the encoding to only the cache key but don't expose this detail in the entity_tag
-        key = %{#{key}:#{hash_value_str(data[:encoding])}} if data[:encoding] && data[:encoding] != "gzip"
+        key = %{#{key}:#{hash_value_str(data[:encoding])}} if data[:encoding]
```

Comment:
The old dirty keys do not have a `content-encoding` field in the header, which is the reason we saw so many `no implicit conversion of nil into String` errors. This PR just simply add `gzip` to the cache key, so all the old version of keys are invalidated and expired in one hour.

The code change: https://github.com/Shopify/response_bank/pull/67/files#diff-c566048d10900a7b76aacf5d9ebe723da577dd792abe6aaa5f3cf39e3620dc0bL42

Reply:
is there any code path where encoding will be nil?